AmplifyPlugins/Core/AWSPluginsCore/Model/GraphQLRequest/GraphQLRequest+AnyModelWithSync.swift

```diff
@@ -169,15 +173,15 @@ extension GraphQLRequest: ModelSyncGraphQLRequestFactory {
                                     authType: AWSAuthorizationType? = nil) -> GraphQLRequest<MutationSyncResult> {
 
         var documentBuilder = ModelBasedGraphQLDocumentBuilder(modelSchema: modelSchema,
-                                                               operationType: .subscription)
+                                                               operationType: .subscription,
+                                                               primaryKeysOnly: true)
```

Comment:
The `subscription` API in GraphQLRequest+AnyModelWithSync will pass in `primaryKeysOnly` true initially. This is reverted back to false if it detects that the models do not have the lazy loading enabled. When lazy loading is enabled, then the selection set will be reduced to just the primary keys in the nested model, thus supporting mutation events sourced from other platforms. 

While testing Flutter with this backport, DataStore.start() will establish the subscriptions. The first step is to make sure the subscription's selection sets are reduced as expected.